### PR TITLE
ui: Filter synthetic nodes on nodes list page

### PR DESCRIPTION
**Pull request by WenInCode**
### Description
Filters synthetic-nodes (agentless) from the nodes index page. I used the same provider pattern that @LevelbossMike is using for search in [his peer details PR](https://github.com/hashicorp/consul/pull/14947), but based the filtering on the [reject-by](https://github.com/DockYard/ember-composable-helpers/blob/master/addon/helpers/reject-by.js) composable helper.

### Testing & Reproduction steps
~_I set the `synthetic-node` meta property to randomly assign itself either true or false. So you should see nodes get filtered out based on that key. Easy way is to inspect the response from the network request for nodes and compare it to the list of nodes shown on the page._~

Unfortunately, randomly mocking the synthetic nodes seems to be causing other node tests to fail with sigint somewhere. I'd like to look into this further, but due to the code freeze time constraints I have removed the mocking for now. If you wish to test it for yourself, you can add them to the mock-api and run the ui.

```
diff --git a/ui/packages/consul-ui/mock-api/v1/internal/ui/nodes b/ui/packages/consul-ui/mock-api/v1/internal/ui/nodes
index 1ebb42e2c..26a2ac501 100644
--- a/ui/packages/consul-ui/mock-api/v1/internal/ui/nodes
+++ b/ui/packages/consul-ui/mock-api/v1/internal/ui/nodes
@@ -25,6 +25,7 @@
             },
             "Meta": {
               "consul-network-segment":"",
+              "synthetic-node": "${fake.helpers.randomize(["true", "false", "false", "false"])}"
             },
             "Services":[
                 ${
```

### Links
- [NET-911](https://hashicorp.atlassian.net/browse/NET-911)

### Screenshots
Screen Shot 2022-10-12 at 3 11 40 PM
Screen Shot 2022-10-12 at 3 12 13 PM

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [x] not a security concern

### CODE REVIEW

File: ui/packages/consul-ui/tests/acceptance/dc/nodes/index.feature [hunk 1]
```diff
@@ -21,6 +21,31 @@ Feature: dc / nodes / index
     Then the url should be /dc-1/nodes
     Then I see 1 node models
     And I see status on the nodes.0 like "critical"
+  Scenario: Viewing nodes list page should not show synthetic nodes
```
Comment (LevelbossMike) [inline ui/packages/consul-ui/tests/acceptance/dc/nodes/index.feature:24]: 😍

**Review by LevelbossMike — CHANGES_REQUESTED** [review]
Looks great! I added a comment about the introduction of the rejectBy-provider.

File: ui/packages/consul-ui/app/templates/dc/nodes/index.hbs [hunk 2]
```diff
@@ -76,52 +76,54 @@ as |route|>
...
-            />
-          </collection.Collection>
-          <collection.Empty>
-            <EmptyState
-              @login={{route.model.app.login.open}}
-            >
-              <BlockSlot @name="header">
-                <h2>
-                  {{t 'routes.dc.nodes.index.empty.header'
+        <Providers::RejectBy @items={{items}} @path="Meta.synthetic-node" @value="true" as |filtered|>
```
[the remark this answers is not in the document]
Comment (WenInCode) [reply ui/packages/consul-ui/app/templates/dc/nodes/index.hbs:79]: I was actually going to reach for the composable-helper first, but then I thought it had to be used in an `each` block and return the filtered items one by one. Now I'm realizing that is definitely not the case and I can use the reject-by straight up, which I am more than happy to do. 

Appreciate your explanation :)

**Review by LevelbossMike — APPROVED** [review]
:rocket:

**Pull request merged**